src/TransientCleaner.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * Delete WordPress transients on content save, based on user-input configuration.
+ *
+ * @package    studiometa/wp-toolkit
+ * @author     Studio Meta <agence@studiometa.fr>
+ * @copyright  2020 Studio Meta
+ * @license    https://opensource.org/licenses/MIT
+ * @since      1.0.0
+ * @version    1.0.0
+ */
+
+namespace Studiometa\WPToolkit;
+
+/**
+ * TransientCleaner class.
+ */
+class TransientCleaner {
+	const PREFIX                   = 'wp_transient_cleaner_';
+	const OPTION_STORED_TRANSIENTS = self::PREFIX . 'stored_transients';
+
+	/**
+	 * Class instance
+	 *
+	 * @var object|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Transients configuration.
+	 *
+	 * @var array
+	 */
+	private $config = array();
+
+	/**
+	 * Stored transients.
+	 *
+	 * @var array|bool
+	 */
+	private $stored_transients;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array $config Configuration.
+	 */
+	public function __construct( $config = array() ) {
+		$this->set_stored_transients( get_option( self::OPTION_STORED_TRANSIENTS ) );
+		$this->set_config( $config );
+		$this->define_public_hooks();
+	}
+
+	/**
+	 * Get class instance.
+	 *
+	 * @param array $config Configuration.
+	 *
+	 * @return object Class instance
+	 */
+	public static function get_instance( $config = array() ) {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new TransientCleaner( $config );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize hooks and filters.
+	 *
+	 * @return void
+	 */
+	public function define_public_hooks() {
+		add_action( 'save_post', array( $this, 'post_transient_cleaner' ), 10, 2 );
+		add_action( 'edit_term', array( $this, 'term_transient_cleaner' ), 10, 3 );
+		add_action( 'updated_option', array( $this, 'option_transient_cleaner' ), 10, 2 );
+		add_action( 'setted_transient', array( $this, 'store_transient_key' ) );
+		add_filter( 'pre_update_option_' . self::OPTION_STORED_TRANSIENTS, array( $this, 'merge_stored_transients_option_values' ), 10, 3 );
+		add_filter( 'update_option_' . self::OPTION_STORED_TRANSIENTS, array( $this, 'set_stored_transients' ), 20, 1 );
+	}
+
+	/**
+	 * Get config.
+	 *
+	 * @return array
+	 */
+	public function get_config() {
+		return $this->config;
+	}
+
+	/**
+	 * Set config.
+	 *
+	 * @param array $config Configuration.
```

Comment:
**suggestion:** Add an inline `{@example }` tag for easier usage within our IDE

As the configuration array can be quite complex, having an example of its usage by hovering the method can be helpful. The screenshot below shows how it could be displayed in Sublime Text:

![Capture d’écran 2021-04-07 à 09 57 36](https://user-images.githubusercontent.com/250145/113831344-ca93e000-9787-11eb-9ff0-19c305c69463.png)

Reply:
The docblock I wrote for the screenshot:

```php
	/**
	 * Set config.
	 *
	 * {@example }
	 * ```php
	 * TransientCleaner::get_instance()
	 *   ->set_config( 
	 *     array(
	 *       'post' => array(
	 *         'all' => array(
	 *           TransientCleaner::PREFIX . 'transient_key',
	 *         ),
	 *         'post_type_key' => array(
	 *           TransientCleaner::PREFIX . 'transient_key',
	 *           TransientCleaner::PREFIX . 'transient_key_1',
	 *         )
	 *       ),
	 *       'term' => array(
	 *         'all'                    => array(),
	 *         'your_taxonomy_type_key' => array(),
	 *         'category'               => array(),
	 *       ),
	 *       'option' => array(
	 *         'all'             => array(),
	 *         'option_key'      => array(),
	 *         'blogdescription' => array(),
	 *       ),
	 *     )
	 *   );
	 * ```
	 * 
	 * @param array $config Configuration.
	 *
	 * @return TransientCleaner
	 */
	public function set_config( array $config ) {
		$this->config = $config;

		return $this;
	}
```